.github/workflows/php-test-plugins.yml

```diff
@@ -71,17 +73,38 @@ jobs:
         run: npm run wp-env start
       - name: Composer Install
         run: npm run wp-env run tests-cli -- --env-cwd="wp-content/plugins/$(basename $(pwd))" composer install --no-interaction --no-progress
+      - name: Update Composer Dependencies
+        run: composer update --with-all-dependencies --no-interaction --no-progress
+      - name: Install PHPUnit
+        run: composer require --dev phpunit/phpunit:${{ matrix.phpunit }}
+        if: matrix.phpunit != ''
       - name: Running single site unit tests
         run: |
           if [ "${{ matrix.coverage }}" == "true" ]; then
-            npm run test-php -- --coverage-clover=coverage-${{ github.sha }}.xml
+            npm run test-php:performance-lab -- -- -- --coverage-clover=coverage-performance-lab-${{ github.sha }}.xml
+            npm run test-php:auto-sizes -- -- -- --coverage-clover=coverage-auto-sizes-${{ github.sha }}.xml
+            npm run test-php:dominant-color-images -- -- -- --coverage-clover=coverage-dominant-color-images-${{ github.sha }}.xml
+            npm run test-php:embed-optimizer -- -- -- --coverage-clover=coverage-embed-optimizer-${{ github.sha }}.xml
+            npm run test-php:image-prioritizer -- -- -- --coverage-clover=coverage-image-prioritizer-${{ github.sha }}.xml
+            npm run test-php:optimization-detective -- -- -- --coverage-clover=coverage-optimization-detective-${{ github.sha }}.xml
+            npm run test-php:speculation-rules -- -- -- --coverage-clover=coverage-speculation-rules-${{ github.sha }}.xml
+            npm run test-php:web-worker-offloading -- -- -- --coverage-clover=coverage-web-worker-offloading-${{ github.sha }}.xml
+            npm run test-php:webp-uploads -- -- -- --coverage-clover=coverage-webp-uploads-${{ github.sha }}.xml
```

Comment:
@sarthak-19 What's the use case behind including SHA here?

Reply:
I added that as a [suggestion](https://github.com/WordPress/performance/pull/1586#discussion_r1834718375 ) by [felixarntz](https://github.com/felixarntz), but removed since it was causing error in file upload to codecov.